### PR TITLE
Quo2 Wallet: Token Value

### LINKED ISSUE #16833: Quo2 Wallet: Token Value
Implement this component in the quo2 component library as per the design spec.

Reuse Quo2 components wherever applicable.

Designs
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=114-5860&mode=design&t=nPMy2jF6iGVTtuza-0

Acceptance Criteria:

component spec tests - https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md
quo2 preview screen
export from quo2/core.cljs
following best practices - https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/16833

This PR implements Token Value list item component.

[Designs.](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=114-5860&mode=design&t=nPMy2jF6iGVTtuza-0)

Preview:
IMG-3408
IMG-3409

### CODE REVIEW

File: src/quo2/components/list_items/token_value/component_spec.cljs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                :color    :blue
+                :metrics? true
+                :values   {:crypto-value      "0.00"
+                           :fiat-value        "€0.00"
+                           :percentage-change "0.00"
+                           :fiat-change       "€0.00"}}])
+    (h/is-truthy (h/get-by-label-text :arrow-icon))))
+
+
+
```
Comment (smohamedjavid) [inline src/quo2/components/list_items/token_value/component_spec.cljs]: Extra lines

File: src/status_im2/contexts/quo_preview/list_items/token_value.cljs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+               :value "Negative"}]}
+   {:label   "Color:"
+    :key     :color
+    :type    :select
+    :options [{:key   :blue
+               :value "Blue"}
+              {:key   :orange
+               :value "Orange"}
+              {:key   :green
+               :value "Green"}]}
```
Comment (smohamedjavid) [inline src/status_im2/contexts/quo_preview/list_items/token_value.cljs]: I believe this will be the customization color. WDYT?

**Review by smohamedjavid — COMMENTED** [review]
Nice work! @OmarBasem

File: src/status_im2/contexts/quo_preview/list_items/token_value.cljs (hunk 2 above)
Comment (OmarBasem) [reply src/status_im2/contexts/quo_preview/list_items/token_value.cljs]: @smohamedjavid Oh I didn't know we have a preview option for that. Thanks.

**Comment by OmarBasem** [comment]
Hey @Francesca-G,

a design review please :)

Component is under list-items category

File: src/status_im2/contexts/quo_preview/list_items/token_value.cljs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   {:label "Metrics?:"
+    :key   :metrics?
+    :type  :boolean}])
+
+(defn preview
+  []
+  (let [state (reagent/atom {:token    :snt
+                             :state    :default
+                             :status   :empty
+                             :color    :blue
```
Comment (smohamedjavid) [inline src/status_im2/contexts/quo_preview/list_items/token_value.cljs]: We need to replace the `color` key with the `customization-color` key (and inside the component too) as it's the standard key across the codebase.

Additionally,  the `(preview/customization-color-option)` uses `customization-color` key. Any changes in the color will not be reflected in the component.

Comment (OmarBasem) [reply src/status_im2/contexts/quo_preview/list_items/token_value.cljs]: Fixed it. Thanks @smohamedjavid 👍

File: src/quo2/components/list_items/token_value/view.cljs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    [quo2.components.markdown.text :as text]
+    [quo2.foundations.colors :as colors]
+    [quo2.foundations.resources :as resources]
+    [quo2.theme :as quo.theme]
+    [react-native.core :as rn]
+    [quo2.foundations.common :as common]
+    [quo2.components.list-items.token-value.style :as style]))
+
+(defn- internal-view
+  [{:keys [theme customization-color state status token metrics? values]}]
```
Comment (J-Son89) [inline src/quo2/components/list_items/token_value/view.cljs:14]: let's refer to 'color' as 'customisation-color' as we use it that way for every other component

**Review by Francesca-G — COMMENTED** [review]
Here's the[ Figma Frame ](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3374%3A300992&mode=design&t=e0h5uHl864zfXoWK-1)with the review

**Review by Francesca-G — APPROVED** [review]

File: src/quo2/core.cljs [hunk 5]
```diff
@@ -58,6 +58,7 @@
     quo2.components.list-items.menu-item
     quo2.components.list-items.preview-list
     quo2.components.list-items.user-list
+    quo2.components.list-items.token-value.view
```
Comment (J-Son89) [inline src/quo2/core.cljs:60]: nit: let's keep this file alpha sorted for imports and exports 👍

File: src/quo2/core_spec.cljs [hunk 6]
```diff
@@ -52,4 +52,5 @@
     [quo2.components.wallet.network-amount.component-spec]
     [quo2.components.wallet.network-bridge.component-spec]
     [quo2.components.wallet.account-card.component-spec]
-    [quo2.components.wallet.token-input.component-spec]))
+    [quo2.components.wallet.token-input.component-spec]
```
Comment (J-Son89) [inline src/quo2/core_spec.cljs]: nit: let's keep this file alpha sorted for imports and exports 👍

**Review by J-Son89 — APPROVED** [review]
nice one @OmarBasem! 🚀

**Pull request merged**